core/src/main/scala/kafka/migration/MigrationPropagator.scala

```diff
@@ -79,6 +81,18 @@ class MigrationPropagator(
     _image = image
   }
 
+  /**
+   * A very expensive function that creates a map with an entry for every partition that exists, from
+   * (topic name, partition index) to partition registration.
+   */
+  def materializePartitions(topicsImage: TopicsImage): util.Map[TopicPartition, PartitionRegistration] = {
```

Comment:
Curious - what is the reason for using `util.Map` instead of one of scala map DS?

Reply:
Using mutable Scala maps gives a bunch of type errors (which I'm sure are fixable, but using a Java map is simple and straightforward)

Reply:
Thanks.